internal/server/openapi.go

```diff
@@ -252,56 +249,33 @@ func setTagInfo(f reflect.StructField, t, parent reflect.Type, schema, parentSch
 
 var exampleTime = time.Date(2022, 3, 14, 9, 48, 0, 0, time.UTC).Format(time.RFC3339)
 
+type describeSchema interface {
+	DescribeSchema(schema *openapi3.Schema)
+}
+
 // `type` can be one of the following only: "object", "array", "string", "number", "integer", "boolean", "null".
 // `format` has a few defined types, but can be anything. https://swagger.io/docs/specification/data-models/data-types/
 func setTypeInfo(t reflect.Type, schema *openapi3.Schema) {
-	switch structNameWithPkg(t) {
-	case "api.Time", "time.Time":
-		schema.Type = "string"
-		schema.Format = "date-time" // date-time is rfc3339
-		schema.Example = exampleTime
-		if len(schema.Description) == 0 {
-			schema.Description = "formatted as an RFC3339 date-time"
-		}
-		return
-
-	case "api.Duration", "time.Duration":
-		schema.Type = "string"
-		schema.Format = "duration"
-		schema.Example = "72h3m6.5s"
-		if len(schema.Description) == 0 {
-			schema.Description = "a duration of time supporting (h)ours, (m)inutes, and (s)econds"
-		}
-		return
-
-	case "uid.ID":
-		schema.Type = "string"
-		schema.Format = "uid"
-		schema.Pattern = `[\da-zA-HJ-NP-Z]{1,11}`
-		schema.Example = "4yJ3n3D8E2"
-		return
-
-	case "api.IDOrSelf":
-		schema.Type = "string"
-		schema.Format = "uid|self"
-		schema.Pattern = `[\da-zA-HJ-NP-Z]{1,11}|self`
-		schema.Example = "4yJ3n3D8E2"
-		schema.Description = "a uid or the literal self"
+	// TODO: convert to value earlier?
+	value := reflect.New(t).Interface()
+	if ds, ok := value.(describeSchema); ok {
+		ds.DescribeSchema(schema)
 		return
+	}
 
-	case "uid.PolymorphicID":
-		schema.Type = "string"
-		schema.Format = "poly-uid"
-		schema.Pattern = `\w:[\da-zA-HJ-NP-Z]{1,11}`
-		schema.Example = "i:4yJ3n3D8E3"
+	switch value.(type) {
+	case time.Time:
+		panic("field must use api.Time")
+	case time.Duration:
+		panic("field must use api.Duration")
+	}
```

Comment:
These are new. Now that we have our own API types for time and duration we should warn a contributor if they accidentally use the ones from stdlib (instead of supporting the stdlib types which would make our API inconsistent).